rules/directive-restrict.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+module.exports = function(context) {
+    var utils = require('./utils/utils');
+
+    var options = context.options[0] || {};
+    var restrictOpt = options.restrict || 'AE';
+    var explicitRestrict = options.explicit === 'always';
+    var restrictChars = [];
+
+    restrictOpt.split('').forEach(function(char) {
+        if ('ACEM'.indexOf(char) === -1) {
```

Comment:
This code is unnecessary, because you already check the options with the json schema. That's also the reason why the code coverage decreases.

var restrictedChars = restrictOps.split('');